src/apps/frontend/types/async-operation.ts

```diff
@@ -0,0 +1,11 @@
+import { ApiError } from './service-response';
+
+export interface AsyncResult<T, AsyncError> {
+  error?: AsyncError;
+  isLoading: boolean;
+  result?: T;
+}
+
+export interface UseAsyncResponse<T> extends AsyncResult<T, ApiError> {
```

Comment:
This is a high level construct, should not anything about api.

Reply:
Done